src/pepr/prometheus/index.ts

```diff
@@ -25,114 +25,121 @@ export const prometheus = new Capability({
 
-    // Add an exempt scrape class if explicitly opted out via annotation OR targeting a non-istio-injected namespace
-    if (pm.Raw.metadata?.annotations?.["uds/skip-mutate"] || !(await isIstioInjected(pm))) {
+    const sc = pm.Raw.spec!.scrapeClass;
+    if (sc !== undefined && sc !== "istio-certs" && sc !== "exempt") {
       log.info(
-        `Mutating scrapeClass to exempt PodMonitor ${pm.Raw.metadata?.name} from default scrapeClass mTLS config`,
+        `PodMonitor ${pm.Raw.metadata?.name} uses custom scrapeClass (${sc}); skipping mutation.`,
       );
-      pm.Raw.spec.scrapeClass = "exempt";
-      // Support the legacy (Prometheus 2.x fallback) until upstream applications properly handle protocol
-      if (!pm.Raw.spec.fallbackScrapeProtocol) {
-        pm.Raw.spec.fallbackScrapeProtocol = FallbackScrapeProtocol.PrometheusText004;
-      }
-
       return;
-    } else {
-      log.info(`Patching pod monitor ${pm.Raw.metadata?.name} for mTLS metrics`);
-      const endpoints: PodMonitorEndpoint[] = pm.Raw.spec.podMetricsEndpoints || [];
-      endpoints.forEach(endpoint => {
-        endpoint.scheme = PodMonitorScheme.HTTPS;
-      });
-      pm.Raw.spec.podMetricsEndpoints = endpoints;
-      // Support the legacy (Prometheus 2.x fallback) until upstream applications properly handle protocol
-      if (!pm.Raw.spec.fallbackScrapeProtocol) {
-        pm.Raw.spec.fallbackScrapeProtocol = FallbackScrapeProtocol.PrometheusText004;
-      }
     }
-  });
 
-// This assumes istio-injection === strict mTLS due to complexity around mTLS lookup
-async function isIstioInjected(monitor: PrometheusServiceMonitor | PrometheusPodMonitor) {
-  // If monitor allows any namespace assume istio injection
-  if (monitor.Raw.spec?.namespaceSelector?.any) {
-    return true;
-  }
-
-  const namespaces = monitor.Raw.spec?.namespaceSelector?.matchNames || [
-      monitor.Raw.metadata?.namespace,
-    ] || ["default"];
-
-  for (const ns of namespaces) {
-    const namespace = await K8s(kind.Namespace).Get(ns);
-    if (namespace.metadata?.labels && namespace.metadata.labels["istio-injection"] === "enabled") {
-      return true;
+    if (
+      pm.Raw.metadata?.annotations?.["uds/skip-mutate"] ||
+      !(await isIstioInjected(pm)) ||
+      sc === "exempt"
+    ) {
+      log.info(`PodMonitor ${pm.Raw.metadata?.name} meets skip conditions; clearing scrapeClass.`);
+      delete pm.Raw.spec!.scrapeClass;
+      return;
     }
-  }
 
-  return false;
-}
+    log.info(
+      `Patching PodMonitor ${pm.Raw.metadata?.name}: clearing scrapeClass, setting endpoints to HTTP, and removing TLS config.`,
+    );
+    delete pm.Raw.spec!.scrapeClass;
+    if (pm.Raw.spec?.podMetricsEndpoints && Array.isArray(pm.Raw.spec.podMetricsEndpoints)) {
+      pm.Raw.spec.podMetricsEndpoints.forEach((endpoint: PodMonitorEndpoint) => {
+        endpoint.scheme = PodMonitorScheme.HTTP;
+        if (endpoint.tlsConfig) {
+          delete endpoint.tlsConfig;
+        }
```

Comment:
```suggestion
    log.info(
      `Patching PodMonitor ${pm.Raw.metadata?.name}: clearing scrapeClass and setting endpoints to HTTP.`,
    );
    delete pm.Raw.spec!.scrapeClass;
    if (pm.Raw.spec?.podMetricsEndpoints && Array.isArray(pm.Raw.spec.podMetricsEndpoints)) {
      pm.Raw.spec.podMetricsEndpoints.forEach((endpoint: PodMonitorEndpoint) => {
        endpoint.scheme = PodMonitorScheme.HTTP;
```

While this is unlikely to cause issues, our "sidecar mutations" [never actually set the tlsConfig for podmonitors](https://github.com/defenseunicorns/uds-core/blob/main/src/pepr/prometheus/index.ts#L105-L116). If one is present, that means the end user specified it and it should probably stay as is.